src/Antiforgery/src/Internal/DefaultAntiforgery.cs

```diff
@@ -100,35 +98,43 @@ public async Task<bool> IsRequestValidAsync(HttpContext httpContext)
             throw new ArgumentNullException(nameof(httpContext));
         }
 
+        var (result, _) = await TryValidateAsync(httpContext, validateIdempotentRequests: false);
+        return result;
+    }
+
+    internal async ValueTask<(bool success, string? errorMessage)> TryValidateAsync(HttpContext httpContext, bool validateIdempotentRequests)
+    {
         CheckSSLConfig(httpContext);
 
         var method = httpContext.Request.Method;
-        if (HttpMethods.IsGet(method) ||
+        if (
+            !validateIdempotentRequests &&
+            (HttpMethods.IsGet(method) ||
             HttpMethods.IsHead(method) ||
             HttpMethods.IsOptions(method) ||
-            HttpMethods.IsTrace(method))
+            HttpMethods.IsTrace(method)))
         {
             // Validation not needed for these request types.
-            return true;
+            return (true, null);
         }
 
         var tokens = await _tokenStore.GetRequestTokensAsync(httpContext);
```

Comment:
Something should catch the AntiforgeryValidationException and convert it to a 400.
https://github.com/dotnet/aspnetcore/blob/a450cb69b5e4549f5515cdb057a68771f56cefd7/src/Antiforgery/src/Internal/DefaultAntiforgeryTokenStore.cs#L62-L77